### ISSUE TITLE
Discord Invalid OAuth2 redirect_uri

### ISSUE THREAD

**Issue by Caleb-Cohen** [Caleb-Cohen/Together#60]
<img width="707" alt="Screen Shot 2022-11-14 at 6 12 37 PM" src="https://user-images.githubusercontent.com/75268174/201794481-1e8cee4b-4dc7-42b3-aeb1-145089e041b3.png">

**Comment by WinstonHall** [comment]
Unless someone is planning to do a big authentication systems overhaul I would suggest cutting out the functionality that requires env_variables in place to run ex(Login, Logout, Auth etc...), unless someone is making a test environment that could be used by everyone.

**Comment by Caleb-Cohen** [comment]
> Unless someone is planning to do a big authentication systems overhaul I would suggest cutting out the functionality that requires env_variables in place to run ex(Login, Logout, Auth etc...), unless someone is making a test environment that could be used by everyone.

Good point. I think the auth system is important and can build out a test environment. Probably good to just have it regardless as we keep going deeper.

**Comment by WinstonHall** [comment]
In my head I imagine you decouple one major thing, the back-end or front-end. I don't think the front-end will be bottle-necked as bad as it is currently with the required dev environment, which from what I could tell would be me opening another database/using my current mongo uri, and registering an app to discord to get an id and secret. That's not necessarily to bad, but if the other stuff using environment variables is required then I gotta say that's the long route.

**Comment by Caleb-Cohen** [comment]
> In my head I imagine you decouple one major thing, the back-end or front-end. I don't think the front-end will be bottle-necked as bad as it is currently with the required dev environment, which from what I could tell would be me opening another database/using my current mongo uri, and registering an app to discord to get an id and secret. That's not necessarily to bad, but if the other stuff using environment variables is required then I gotta say that's the long route.

Some sort of refractor will be required. I would just like to assess that after MVP. Currently, it's just mongo and discord, and discord is only required if you're helping with the auth portion, but spending some time to make it easier for people to plug into the same dev environment has some value.

**Comment by WinstonHall** [comment]
Yea you wouldn't have to remove anything, honestly its not like the project has to have a group effort/commit pushed forward. Literally anyone with a fork/copy can comment that stuff out and isolate an environment to build out components & style them.

**Comment by WinstonHall** [comment]
I'm not sure when we call this thread kaput and where we carry on these conversation when that happens. To the second half of, what you said @Caleb-Cohen, yea I don't mind the idea of people forking around, I just thought we could at bare minimum have a dedicated branch and set up some more sorta rules on the branches to help prevent/alleviate problems. I think in a way it helps not just protect the code base but give more structure for people who aren't sure "where and when to what to be doing". I'm game for helping with that stuff I guess I kinda just try to wait for someone to give a nudge before I bring things up, like Next JS the code standards you want for the project. I'm eager to get the ball rolling and sitting back as best I can to let you voice what you want. You started this thing so like I'm just trying not to step on anyone's toe's.

**Comment by Caleb-Cohen** [comment]
> I'm not sure when we call this thread kaput and where we carry on these conversation when that happens.

Oops, sorry! I closed the Issue I fixed the Oauth redirect. 

To carry on discussions like refactoring, next JS code standards, and dedicated branches I think those are better placed in the discussions. I would prefer to avoid bringing up more abstract problems that may require deeper discussions in the issues tab as I would prefer to open and close these issues more quickly. That said we do have the beginnings of a test environment in issue #55.
